base/inference.jl

```diff
@@ -191,7 +193,7 @@ mutable struct InferenceState
                     vararg_type = rewrap(vararg_type, linfo.specTypes)
                 end
                 s_types[1][la] = VarState(vararg_type, false)
-                src.slottypes[la] = widenconst(vararg_type)
+                src.slottypes[la] = vararg_type
```

Comment:
For slottypes in general that's true, but this was causing arguments to be given e.g. type `Void` instead of `Const(nothing)`. Ideally we'd treat those the same anyway, but this was an easier way to achieve that.